plugins/modules/ec2_transit_gateway_info.py

```diff
@@ -168,71 +159,77 @@
             sample: "tgw-02c42332e6b7da829"
 """
 
-try:
-    import botocore
-except ImportError:
-    pass  # handled by imported AnsibleAWSModule
+from typing import Any
+from typing import Dict
+from typing import List
 
 from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
 
-from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
-from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AnsibleEC2Error
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import describe_ec2_transit_gateways
 from ansible_collections.amazon.aws.plugins.module_utils.tagging import boto3_tag_list_to_ansible_dict
 from ansible_collections.amazon.aws.plugins.module_utils.transformation import ansible_dict_to_boto3_filter_list
 
 from ansible_collections.community.aws.plugins.module_utils.modules import AnsibleCommunityAWSModule as AnsibleAWSModule
 
 
-class AnsibleEc2TgwInfo(object):
-    def __init__(self, module, results):
-        self._module = module
-        self._results = results
-        self._connection = self._module.client("ec2")
-        self._check_mode = self._module.check_mode
+def get_transit_gateway_response(module: AnsibleAWSModule, connection) -> Dict[str, Any]:
+    """
+    Get transit gateway response from AWS.
+
+    module     : AnsibleAWSModule object
+    connection : boto3 client connection object
+    :return: Response from describe_transit_gateways call
+    """
+    filters = ansible_dict_to_boto3_filter_list(module.params["filters"])
+    transit_gateway_ids = module.params["transit_gateway_ids"]
 
-    @AWSRetry.exponential_backoff()
-    def describe_transit_gateways(self):
-        """
-        Describe transit gateways.
+    params = {}
+    if transit_gateway_ids:
+        params["TransitGatewayIds"] = transit_gateway_ids
+    if filters:
+        params["Filters"] = filters
 
-        module  : AnsibleAWSModule object
-        connection  : boto3 client connection object
-        """
-        # collect parameters
-        filters = ansible_dict_to_boto3_filter_list(self._module.params["filters"])
-        transit_gateway_ids = self._module.params["transit_gateway_ids"]
+    try:
+        result = describe_ec2_transit_gateways(connection, **params)
+        return result if result else {"TransitGateways": []}
+    except AnsibleEC2Error as e:
```

Comment:
This try-except block is probably not necessary because one has already been used in the main

Reply:
removed